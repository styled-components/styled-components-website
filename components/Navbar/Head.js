import styled, { css } from 'styled-components'

import rem from '../../utils/rem'
import { mobile } from '../../utils/media'

const Head = styled.div`
  height: ${rem(300)};
  padding: ${rem(65)} ${rem(85)} ${rem(45)};

  ${mobile(css`
    display: flex;
    flex-direction: row;
    align-items: center;

    height: ${rem(70)};
    padding: 0 ${rem(15)};
  `)}
`

export default Head
