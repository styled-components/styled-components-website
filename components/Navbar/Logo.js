import styled, { css } from 'styled-components'

import rem from '../../utils/rem'
import { mobile } from '../../utils/media'

const Logo = styled.div`
  display: inline-block;
  width: 100%;
  height: 100%;

  box-sizing: border-box;

  background-image: url(/static/logo.png);
  background-size: contain;
  background-position: center;

  ${mobile(css`
    background-image: url(/static/icon.png);
    background-position: center;
    width: ${rem(70)};
    height: ${rem(70)};
  `)}
`

export default Logo
