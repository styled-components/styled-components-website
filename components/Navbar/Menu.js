import styled, { css } from 'styled-components'

import rem from '../../utils/rem'
import Link, { StyledLink } from '../Link'
import titleToDash from '../../utils/titleToDash'
import { lightGrey } from '../../utils/colors'
import { mobile } from '../../utils/media'

import { pages } from '../../pages/docs.json'

const MenuOuter = styled.aside`
  display: block;
  box-sizing: border-box;

  ${mobile(css`
    height: ${p => p.isFolded ? '0' : `calc(100vh - ${rem(70)})`};
    transition: height .2s ease-in-out;
    overflow: hidden;
  `)}
`

const MenuInner = styled.div`
  display: block;
  box-sizing: border-box;
  height: 100%;

  ${mobile(css`
    border-top: 2px solid ${lightGrey};
    padding-top: ${rem(10)};
    padding-bottom: ${rem(60)};
    overflow-y: scroll;
  `)}
`

const Section = styled.div`
  margin-bottom: ${rem(30)};
`

const SectionTitle = styled.h4`
  display: block;
  margin: ${rem(10)} ${rem(40)};
  font-weight: normal;
`

const SubSection = styled.h5`
  display: block;
  margin: ${rem(10)} ${rem(40)} ${rem(10)} ${rem(55)};
  font-size: 0.9rem;
  font-weight: normal;
`

const Menu = ({ isFolded, onRouteChange }) => (
  <MenuOuter isFolded={isFolded}>
    <MenuInner>
      {
        pages.map(({ title, pathname, sections }) => (
          <Section key={title} onClick={onRouteChange}>
            <SectionTitle>
              <Link href={`/docs/${pathname}`}>
                {title}
              </Link>
            </SectionTitle>

            {
              sections.map(({ title }) => (
                <SubSection key={title}>
                  <StyledLink href={`/docs/${pathname}#${titleToDash(title)}`}>
                    {title}
                  </StyledLink>
                </SubSection>
              ))
            }
          </Section>
        ))
      }
    </MenuInner>
  </MenuOuter>
)

export default Menu
