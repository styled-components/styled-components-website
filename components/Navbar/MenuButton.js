import styled, { css } from 'styled-components'
import ThreeBarsIcon from 'react-octicons-svg/dist/ThreeBarsIcon'

import rem from '../../utils/rem'
import { StyledLink } from '../Link'
import { mobile } from '../../utils/media'

const MenuButton = styled(StyledLink)`
  display: none;
  height: 100%;
  padding: ${rem(20)} 0;
  margin: 0 0 0 auto;
  color: white;
  transform-origin: center;

  ${mobile(css`
    display: block;
  `)}

  &:active {
    opacity: 0.8;
    transform: scale(0.9);
  }

  &:hover {
    background: none;
  }
`

const MenuIcon = styled(ThreeBarsIcon).attrs({
  height: null,
  width: null
})`
  height: ${rem(30)};
`

export default props => (
  <MenuButton {...props}>
    <MenuIcon />
  </MenuButton>
)

