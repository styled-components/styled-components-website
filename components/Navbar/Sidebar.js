import styled, { css } from 'styled-components'

import rem from '../../utils/rem'
import { mobile } from '../../utils/media'
import { violetRed, gold } from '../../utils/colors'
import { headerFont } from '../../utils/fonts'
import captureScroll from '../CaptureScroll'

const Sidebar = styled.nav`
  position: fixed;
  transform: translateZ(0);
  display: block;
  z-index: 1;
  font-family: ${headerFont};

  left: 0;
  top: 0;
  bottom: 0;
  right: auto;

  width: ${rem(300)};
  height: 100%;
  background: linear-gradient(20deg, ${violetRed}, ${gold});
  box-sizing: border-box;
  color: white;
  overflow-y: scroll;

  ${mobile(css`
    bottom: auto;
    right: 0;
    height: auto;
    width: 100%;
    overflow-y: visible;
  `)}
`

export default captureScroll(Sidebar)
