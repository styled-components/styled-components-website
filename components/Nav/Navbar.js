import React, { PureComponent, createRef } from 'react';
import styled, { css } from 'styled-components';
import { Close } from '@styled-icons/material';
import rem from '../../utils/rem';
import { blmGrey, paleGrey } from '../../utils/colors';
import { navbarHeight } from '../../utils/sizes';
import { headerFont } from '../../utils/fonts';
import { mobile } from '../../utils/media';
import Link from '../Link';
import NavLinks from './NavLinks';
import Social from './Social';
import Logo from './Logo';
import MobileNavbar from './MobileNavbar';
import SearchWithAlgolia from './SearchWithAlgolia';

const Wrapper = styled.nav`
  position: fixed;
  left: 0;
  box-sizing: border-box;
  z-index: 3;

  width: 100%;
  height: ${rem(navbarHeight)};

  font-family: ${headerFont};
  font-size: ${rem(15)};
  font-weight: 500;
  background: ${(props) => (props.transparent ? 'transparent' : blmGrey)};
  transition: background 300ms ease-out;
  color: white;
  padding: 0;
`;

const StartWrapper = styled.div`
  display: flex;
  align-items: center;
  justify-content: flex-start;
`;

const EndWrapper = styled.div`
  display: flex;
  align-items: center;
  justify-content: flex-end;
`;
/* stylelint-disable */
const StyledSocial = styled(Social)``;
/* stylelint-enable */

const NormalNavbar = styled.div`
  display: flex;
  align-items: center;
  padding: 0 ${rem(20)};
  justify-content: space-between;
  ${StartWrapper}, ${EndWrapper} ${StyledSocial} {
    ${mobile(css`
      display: none;
    `)};
  }
`;

const StyledModalCloseIcon = styled(Close)`
  width: ${rem(26)};
  height: ${rem(26)};
  color: white;
`;

const AlgoliaModal = styled.div`
  ${mobile(css`
    background: currentColor;
    overflow: auto;
    position: absolute;
    top: 0;
    left: 0;
    right: 0;
  `)};
`;

const baseZ = 1;

const AlgoliaModalHeader = styled.div`
  display: none;
  color: currentColor;

  ${mobile(css`
    display: ${(props) => (props.isOpen ? 'block' : 'none')};

    button {
      cursor: pointer;
      padding: 0;
      position: fixed;
      right: ${rem(10)};
      top: ${rem(11)};
      border: none;
      z-index: ${baseZ + 1};
    }
  `)};
`;

const AlgoliaModalOverlay = styled.div`
  margin-right: ${rem(10)};

  ${mobile(css`
    position: fixed;
    top: 0;
    bottom: 0;
    left: 0;
    right: 0;
    z-index: ${baseZ};
    left: 0;
    display: ${(props) => (props.isOpen ? 'block' : 'none')};
    background: ${paleGrey};
    overflow-y: auto;
    margin: 0;

    .algolia-autocomplete .ds-dropdown-menu {
      position: static !important;
      display: block;
      max-width: 100%;
      min-width: 0;
    }
  `)};
`;

class ModalContainer extends PureComponent {
  modalElement = createRef();
  onModalOverlayClick = (e) => {
    if (!this.modalElement.current.contains(e.target)) {
      this.props.requestModalClose();
    }
  };
  onCloseButtonClick = (e) => {
    e.stopPropagation();
    this.props.requestModalClose();
  };
  render() {
    return (
      <>
        <AlgoliaModalHeader isOpen={this.props.isOpen}>
          <button onClick={this.onCloseButtonClick}>
            <StyledModalCloseIcon />
          </button>
        </AlgoliaModalHeader>
        <AlgoliaModalOverlay onClick={this.onModalOverlayClick} isOpen={this.props.isOpen}>
          <AlgoliaModal ref={this.modalElement}>
            <div>{this.props.children}</div>
          </AlgoliaModal>
        </AlgoliaModalOverlay>
      </>
    );
  }
}

const LogoLink = styled(Link).attrs((/* props */) => ({
  unstyled: true,
  href: '/',
}))`
  display: inline-block;
  vertical-align: center;
  margin-right: ${rem(35)};
`;

class Navbar extends PureComponent {
  state = {
    isOpen: false,
  };
  openModal = () => this.setState(() => ({ isOpen: true }));
  closeModal = () => this.setState(({ isOpen }) => (isOpen ? { isOpen: false } : null));
  render() {
    const { onSideToggle, onMobileNavToggle, isSideFolded, isMobileNavFolded, showSideNav, transparent } = this.props;

    return (
      <Wrapper transparent={transparent}>
        <NormalNavbar>
          <StartWrapper>
            <LogoLink>
              <Logo />
            </LogoLink>
            <NavLinks />
          </StartWrapper>
          <EndWrapper>
            <ModalContainer isOpen={this.state.isOpen} requestModalClose={this.closeModal}>
              <SearchWithAlgolia requestModalClose={this.closeModal} />
            </ModalContainer>
            <StyledSocial />
          </EndWrapper>
        </NormalNavbar>
        <MobileNavbar
          isSideFolded={isSideFolded}
          onSearchButtonClick={this.openModal}
          isMobileNavFolded={isMobileNavFolded}
          onSideToggle={onSideToggle}
          onMobileNavToggle={onMobileNavToggle}
          showSideNav={showSideNav}
        />
      </Wrapper>
    );
  }
}

export default Navbar;
