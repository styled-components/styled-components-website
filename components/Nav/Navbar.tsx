import { Close } from '@styled-icons/material';
import { PureComponent, createRef } from 'react';
import styled, { css } from 'styled-components';
import { blmGrey, paleGrey } from '../../utils/colors';
import { headerFont } from '../../utils/fonts';
import { mobile } from '../../utils/media';
import rem from '../../utils/rem';
import { navbarHeight } from '../../utils/sizes';
import Link from '../Link';
import Logo from './Logo';
import MobileNavbar from './MobileNavbar';
import NavLinks from './NavLinks';
import SearchWithAlgolia from './SearchWithAlgolia';
import Social from './Social';

const Wrapper = styled.nav<{ $transparent?: boolean }>`
  align-items: center;
  background-color: rgba(12, 13, 15, 0.7);
  backdrop-filter: blur(5px);
  box-sizing: border-box;
  color: white;
  display: flex;
  flex-wrap: wrap;
  font-family: ${headerFont};
  font-size: ${rem(15)};
  font-weight: 500;
  justify-content: center;
  height: ${rem(navbarHeight)};
  left: 0;
  padding: 0;
  position: fixed;
  transition: background 300ms ease-out;
  width: 100%;
  z-index: 3;
`;

const StartWrapper = styled.div`
  display: flex;
  align-items: center;
  justify-content: flex-start;
`;

const EndWrapper = styled.div`
  display: flex;
  align-items: center;
  justify-content: flex-end;
`;
/* stylelint-disable */
const StyledSocial = styled(Social)``;
/* stylelint-enable */

const NormalNavbar = styled.div`
  display: flex;
  align-items: center;
  padding: 0 ${rem(20)};
  justify-content: space-between;

  ${StartWrapper}, ${EndWrapper} ${StyledSocial} {
    ${mobile(css`
      display: none;
    `)};
  }
`;

const StyledModalCloseIcon = styled(Close)`
  width: ${rem(26)};
  height: ${rem(26)};
  color: white;
`;

const AlgoliaModal = styled.div`
  ${mobile(css`
    background: currentColor;
    overflow: auto;
    position: absolute;
    top: 0;
    left: 0;
    right: 0;
  `)};
`;

const baseZ = 1;

const AlgoliaModalHeader = styled.div<{ $isOpen?: boolean }>`
  display: none;
  color: currentColor;

  ${mobile(css<{ $isOpen?: boolean }>`
    display: ${props => (props.$isOpen ? 'block' : 'none')};

    button {
      cursor: pointer;
      padding: 0;
      position: fixed;
      right: ${rem(10)};
      top: ${rem(11)};
      border: none;
      z-index: ${baseZ + 1};
    }
  `)};
`;

const AlgoliaModalOverlay = styled.div<{ $isOpen?: boolean }>`
  margin-right: ${rem(10)};

  ${mobile(css<{ $isOpen?: boolean }>`
    position: fixed;
    top: 0;
    bottom: 0;
    left: 0;
    right: 0;
    z-index: ${baseZ};
    left: 0;
    display: ${props => (props.$isOpen ? 'block' : 'none')};
    background: ${paleGrey};
    overflow-y: auto;
    margin: 0;

    .algolia-autocomplete .ds-dropdown-menu {
      position: static !important;
      display: block;
      max-width: 100%;
      min-width: 0;
    }
  `)};
`;

class ModalContainer extends PureComponent<{ isOpen?: boolean; requestModalClose: () => void }> {
  modalElement = createRef<HTMLDivElement>();

  onModalOverlayClick = (e: React.MouseEvent) => {
    if (!this.modalElement.current?.contains(e.target as Node)) {
      this.props.requestModalClose();
    }
  };

  onCloseButtonClick = (e: React.MouseEvent) => {
    e.stopPropagation();
    this.props.requestModalClose();
  };

  render() {
    return (
      <>
        <AlgoliaModalHeader $isOpen={this.props.isOpen}>
          <button onClick={this.onCloseButtonClick}>
            <StyledModalCloseIcon />
          </button>
        </AlgoliaModalHeader>
        <AlgoliaModalOverlay onClick={this.onModalOverlayClick} $isOpen={this.props.isOpen}>
          <AlgoliaModal ref={this.modalElement}>
            <div>{this.props.children}</div>
          </AlgoliaModal>
        </AlgoliaModalOverlay>
      </>
    );
  }
}

const LogoLink = styled(Link).attrs((/* props */) => ({
  unstyled: true,
  href: '/',
}))`
  display: inline-block;
  vertical-align: center;
  margin-right: ${rem(35)};
`;

export interface NavbarProps {
  onSideToggle?: () => void;
  onMobileNavToggle?: () => void;
  isSideFolded?: boolean;
  isMobileNavFolded?: boolean;
  showSideNav?: boolean;
}

function Navbar({ onSideToggle, onMobileNavToggle, isSideFolded, isMobileNavFolded, showSideNav }: NavbarProps) {
  return (
    <Wrapper>
      <MobileNavbar
        isSideFolded={isSideFolded}
        isMobileNavFolded={isMobileNavFolded}
        onSideToggle={onSideToggle}
        onMobileNavToggle={onMobileNavToggle}
        showSideNav={showSideNav}
      />

      <NormalNavbar>
        <StartWrapper>
          <LogoLink aria-label="Styled-components logo">
            <Logo />
          </LogoLink>
          <NavLinks />
        </StartWrapper>
      </NormalNavbar>

      <EndWrapper
        css={css`
          margin-left: auto;
          margin-right: 16px;

          ${mobile(css`
            margin-right: 48px;
          `)}
        `}
      >
        <SearchWithAlgolia />
        <StyledSocial style={{ marginLeft: 16 }} />
      </EndWrapper>
    </Wrapper>
  );
}

export default Navbar;
