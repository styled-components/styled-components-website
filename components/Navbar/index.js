import React, { Component } from 'react'
import Sidebar from './Sidebar'
import Head from './Head'
import Logo from './Logo'
import Text from './Text'
import Menu from './Menu'
import MenuButton from './MenuButton'
import Link from '../Link'

class Navbar extends Component {
  state = {
    isFolded: true
  }

  onFold = () => {
    this.setState({
      isFolded: !this.state.isFolded
    })
  }

  onRouteChange = () => {
    this.setState({ isFolded: true })
  }

  render() {
    const { isFolded } = this.state

    return (
      <Sidebar>
        <Head>
          <Link unstyled href="/docs">
            <Logo />
          </Link>

          <Text>
            <Link unstyled href="/docs">
              styled-components
            </Link>
          </Text>

          <MenuButton onClick={this.onFold} />
        </Head>

        <Menu
          isFolded={isFolded}
          onRouteChange={this.onRouteChange}
        />
      </Sidebar>
    )
  }
}

export default Navbar
