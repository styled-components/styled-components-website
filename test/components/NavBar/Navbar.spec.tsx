import React from 'react';
import renderer from 'react-test-renderer';
import Navbar from '../../../components/Nav/Navbar';

test('Navbar renders correctly', () => {
  const tree = renderer.create(<Navbar />).toJSON();

  expect(tree).toMatchSnapshot();
});
