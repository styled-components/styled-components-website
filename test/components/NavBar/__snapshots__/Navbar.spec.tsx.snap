// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar renders correctly 1`] = `
.c6 {
  background-image: url(/nav-logo.png);
  background-position: center;
  background-size: contain;
  box-sizing: border-box;
  display: inline-block;
  height: 2.2222222222222223rem;
  width: 9.11111111111111rem;
  vertical-align: middle;
}

.c3 {
  background: none;
  outline: none;
  border: none;
  background: transparent;
  cursor: pointer;
  flex: 0 0 auto;
  height: 3.3333333333333335rem;
  padding: 0 0.5555555555555556rem;
  text-align: center;
  transition: all 200ms ease-in-out;
  vertical-align: middle;
}

.c3:hover {
  filter: brightness(0.85);
}

.c3:active {
  filter: brightness(1);
}

.c2+.c2 {
  padding-left: 0;
}

.c4 svg {
  display: inline-block;
}

.c4 svg path {
  fill: currentColor;
}

.c12 {
  flex: 0 0 auto;
  width: 0.2777777777777778rem;
  height: 0.2777777777777778rem;
  margin: 0 0.8333333333333334rem;
  border-radius: 50%;
  background: currentColor;
  opacity: 0.35;
}

.c10 {
  display: flex;
  align-items: center;
  flex: 0 0 auto;
  margin-right: 1.6666666666666667rem;
}

.c11 {
  flex: 0 0 auto;
  display: inline-block;
  line-height: 3.3333333333333335rem;
  transition: opacity 0.2s,transform 0.2s;
  cursor: pointer;
  letter-spacing: 0.022222222222222223rem;
  color: currentColor;
}

.c11:hover,
.c11:focus {
  opacity: 0.8;
}

.c11:active {
  transform: scale(0.95);
  opacity: 0.6;
}

.c1 {
  display: none;
}

.c9 {
  position: absolute;
  top: 3.3333333333333335rem;
  left: 0;
  right: 0;
  height: 3.3333333333333335rem;
  display: flex;
  flex-wrap: nowrap;
  align-items: center;
  justify-content: space-between;
  padding: 0 1.1111111111111112rem;
  transition: height 0.1s;
  user-select: none;
  -webkit-overflow-scrolling: touch;
  overflow-x: scroll;
  overflow-y: hidden;
  background: rgb(33, 33, 33);
  color: #fff;
}

.c5 {
  margin-right: auto;
  transform: translateY(-1px);
}

.c7 {
  transition: transform 0.2s;
  transform-origin: center center;
  transform: rotate(180deg);
}

.c8.c8 {
  width: 2rem;
  height: 2rem;
}

.c17 {
  display: flex;
  align-items: center;
  flex: 1 1 auto;
}

.c18 {
  display: flex;
  margin-right: 1.1111111111111112rem;
  line-height: 3.3333333333333335rem;
  transition: opacity 0.2s,transform 0.2s;
  cursor: pointer;
}

.c18:last-child {
  margin-right: 0;
}

.c18:hover,
.c18:focus {
  opacity: 0.8;
}

.c18:active {
  transform: scale(0.95);
  opacity: 0.6;
}

.c18 svg path {
  fill: currentColor;
}

.c19.c19 {
  width: NaNrem;
  height: 1rem;
}

.c0 {
  align-items: center;
  background-color: rgba(12, 13, 15, 0.7);
  backdrop-filter: blur(5px);
  box-sizing: border-box;
  color: white;
  display: flex;
  flex-wrap: wrap;
  font-family: "Karla",-apple-system,BlinkMacSystemFont,"Segoe UI",Helvetica,Arial,sans-serif,"Apple Color Emoji","Segoe UI Emoji","Segoe UI Symbol";
  font-size: 0.8333333333333334rem;
  font-weight: 500;
  justify-content: center;
  height: 3.3333333333333335rem;
  left: 0;
  padding: 0;
  position: fixed;
  transition: background 300ms ease-out;
  width: 100%;
  z-index: 3;
}

.c14 {
  display: flex;
  align-items: center;
  justify-content: flex-start;
}

.c16 {
  display: flex;
  align-items: center;
  justify-content: flex-end;
}

.c13 {
  display: flex;
  align-items: center;
  padding: 0 1.1111111111111112rem;
  justify-content: space-between;
}

.c15 {
  display: inline-block;
  vertical-align: center;
  margin-right: 1.9444444444444444rem;
}

@media (max-width: 62.5em) {
  .c1 {
    align-items: center;
    display: flex;
    flex-grow: 1;
    height: 3.3333333333333335rem;
    justify-content: space-between;
  }
}

@media (max-width: 62.5em) {

}

<nav
  className="c0"
>
  <div
    className="c1"
  >
    <button
      className="c2 c3"
    >
      <svg
        className="c4"
        height="15"
        width="15"
        xmlns="http://www.w3.org/2000/svg"
        xmlnsXlink="http://www.w3.org/1999/xlink"
      >
        <title>
          close
        </title>
        <use
          fill="#FFF"
          transform="translate(1 1)"
          xlinkHref="#close"
        />
        <defs>
          <path
            d="M-.7.7l13 13 1.4-1.4-13-13L-.7.7zm13-1.4l-13 13 1.4 1.4 13-13-1.4-1.4z"
            id="close"
          />
        </defs>
      </svg>
    </button>
    <a
      aria-label="styled components"
      className="c5"
    >
      <div
        className="c6"
      />
    </a>
    <button
      className="c2 c3"
      style={
        {
          "position": "absolute",
          "right": 0,
        }
      }
    >
      <div
        className="c7"
      >
        <div
          className="c8"
        />
      </div>
    </button>
    <div
      className="c9"
    >
      <nav
        className="c10"
      >
        <a
          className="c11"
        >
          Documentation
        </a>
        <span
          className="c12"
        />
        <a
          className="c11"
        >
          Showcase
        </a>
        <span
          className="c12"
        />
        <a
          className="c11"
        >
          Ecosystem
        </a>
        <span
          className="c12"
        />
        <a
          className="c11"
        >
          Releases
        </a>
      </nav>
    </div>
  </div>
  <div
    className="c13"
  >
    <div
      className="c14"
    >
      <a
        aria-label="Styled-components logo"
        className="c15"
      >
        <div
          className="c6"
        />
      </a>
      <nav
        className="c10"
      >
        <a
          className="c11"
        >
          Documentation
        </a>
        <span
          className="c12"
        />
        <a
          className="c11"
        >
          Showcase
        </a>
        <span
          className="c12"
        />
        <a
          className="c11"
        >
          Ecosystem
        </a>
        <span
          className="c12"
        />
        <a
          className="c11"
        >
          Releases
        </a>
      </nav>
    </div>
  </div>
  <div
    className="c16"
    css={
      [
        "
          margin-left: auto;
          margin-right: 16px;

          ",
        "
  @media (max-width: ",
        "62.5",
        "em) {
    ",
        "
            margin-right: 48px;
          ",
        ";
  }
",
        "
        ",
      ]
    }
  >
    <div
      id="docs-search-input"
    />
    <nav
      className="c17 "
      style={
        {
          "marginLeft": 16,
        }
      }
    >
      <a
        className="c18"
        title="GitHub: Source code"
      >
        <svg
          className="c19"
        />
      </a>
      <a
        className="c18"
        title="Medium: Announcements, blog posts, and more"
      >
        <svg
          className="c19"
        />
      </a>
    </nav>
  </div>
</nav>
`;
